### PR TITLE
add a new config property authentication.ldap.guidKey

### PR DESCRIPTION
**Pull request by lordk911**
for fix issues : https://github.com/NetEase/kyuubi/issues/709

my ldap env use cn but uid for login, uid is the defaut behavior by kyuubi : 
image

then try to connect kyuubi :  
image
image

change kyuubi behavior to use cn:

image

then try to connect kyuubi : 

image

### CODE REVIEW

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 1]
```diff
@@ -314,6 +314,14 @@ object KyuubiConf {
       .stringConf
       .createOptional
 
+  val AUTHENTICATION_LDAP_GUIDKEY: ConfigEntry[String] =
+    buildConf("authentication.ldap.guidKey")
+      .doc("LDAP attribute name whose values are unique in this LDAP server." +
+        "For example:uid or cn.")
+      .version("1.2.0")
```
Comment (ulysses-you) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:321]: do you want to backport this to 1.2.0 ? the master branch now is 1.3.0

Comment (ulysses-you) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:321]: and please update the docs with the cmd:
`KYUUBI_UPDATE=1 build/mvn test -DwildcardSuites=org.apache.kyuubi.config.AllKyuubiConfiguration`

**Comment by ulysses-you** [comment]
thank you for making a PR @lordk911 , welcome to Kyuubi !

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala (hunk 1 above)
Comment (yaooqinn) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:321]: > do you want to backport this to 1.2.0 ? the master branch now is 1.3.0

This is not a big deal, as we haven't decide the final release of kyuubi yet for the process of Apache incubating

Comment (lordk911) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:321]: I've update the docs

**Review by yaooqinn — APPROVED** [review]

**Comment by ulysses-you** [comment]
hi @lordk911 just notice this PR is for `branch-1.2`, can you change the target to master ? thanks

**Comment by yaooqinn** [comment]
It's better to add kshen@gaojihealth.com to your github settings too @lordk911

**Pull request closed without merging**